python/ray/tune/tuner.py

```diff
@@ -122,14 +119,7 @@ def get_dataset():
 
     def __init__(
         self,
-        trainable: Optional[
-            Union[
-                str,
-                Callable,
-                Type[Trainable],
-                "BaseTrainer",
-            ]
-        ] = None,
+        trainable: Optional[TrainableTypeOrTrainer] = None,
```

Comment:
For the user facing API, can we explicitly list out the fully type hint (even though it is verbose)

Reply:
In TunerInternal we can use the shorthand.